src/base/loader/image_file_data.cc

```diff
@@ -67,7 +67,7 @@ doGzipLoad(int fd)
     size_t tmp_len = strlen(P_tmpdir);
     char *tmpnam = (char*) malloc(tmp_len + 20);
     strcpy(tmpnam, P_tmpdir);
-    strcpy(tmpnam+tmp_len, "/gem5-gz-obj-XXXXXX"); // 19 chars
+    strcat(tmpnam, "/gem5-gz-obj-XXXXXX"); // concat 19 chars
```

Comment:
This looks ok to me; shall we though bypass this explicit storage management and rely on std::string instead? We could do string concatenation easily and extract a raw ptr when needed via c_str()

Reply:
I just pushed a patch to utilize `std::string` however to do so I needed to do some funny `strdup` casting to get a `*char` for `mkstemp`. I'm fine either way.